service/src/test/java/uk/gov/hmcts/reform/fpl/service/orders/validator/SupervisionOrderEndDateValidatorTest.java

```diff
@@ -0,0 +1,127 @@
+package uk.gov.hmcts.reform.fpl.service.orders.validator;
+
+import org.junit.jupiter.api.Test;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.event.ManageOrdersEventData;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+import uk.gov.hmcts.reform.fpl.utils.FixedTimeConfiguration;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.List;
+
+import static java.util.Objects.deepEquals;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static uk.gov.hmcts.reform.fpl.enums.orders.SupervisionOrderEndDateType.SET_CALENDAR_DAY;
+import static uk.gov.hmcts.reform.fpl.enums.orders.SupervisionOrderEndDateType.SET_CALENDAR_DAY_AND_TIME;
+import static uk.gov.hmcts.reform.fpl.enums.orders.SupervisionOrderEndDateType.SET_NUMBER_OF_MONTHS;
+import static uk.gov.hmcts.reform.fpl.model.order.OrderQuestionBlock.SUPERVISION_ORDER_END_DATE;
+
+class SupervisionOrderEndDateValidatorTest {
```

Comment:
missing the positive tests and would be good to have boundary tests. For example if you know the precision is at second, you   .manageOrdersSetDateEndDate(dateNow().plusDays(1))) for the negative and  .manageOrdersSetDateEndDate(dateNow())) for the positive (or something along the line....)

Reply:
Excellent point. Ill get into it now